modules/account-lib/src/coin/xrp/transaction.ts

```diff
@@ -1,20 +1,159 @@
-import { BaseTransaction } from '../baseCoin';
+import { BaseCoin as CoinConfig } from '@bitgo/statics';
+import * as rippleTypes from 'ripple-lib/dist/npm/transaction/types';
+import binaryCodec from 'ripple-binary-codec';
+import { BaseTransaction, TransactionType } from '../baseCoin';
 import { BaseKey } from '../baseCoin/iface';
-import { NotImplementedError } from '../baseCoin/errors';
+import { InvalidTransactionError } from '../baseCoin/errors';
+import { SignedXRPTransaction, TxJSON, Signer } from './iface';
+import { KeyPair } from './keyPair';
+import { initApi } from './utils';
 
 export class Transaction extends BaseTransaction {
+  private _xrpTransaction?: rippleTypes.TransactionJSON;
+  private _signedTransaction?: SignedXRPTransaction;
+  private _sender: string;
+
+  constructor(coinConfig: Readonly<CoinConfig>) {
+    super(coinConfig);
+  }
+
+  /**
+   * Get the list of signatures (if any) produced for this transaction.
+   *
+   * @returns {string[]} list of signatures
+   */
+  get signature(): string[] {
+    if (!this._signedTransaction) {
+      return [];
+    }
+    const decodedTx = binaryCodec.decode(this._signedTransaction?.signedTransaction);
+    if (decodedTx.Signers) {
+      const signers = decodedTx.Signers as unknown as { Signer: Signer }[];
+      return signers.map(({ Signer }) => Signer.TxnSignature);
+    }
+    return [];
+  }
+
+  sender(address: string): void {
+    this._sender = address;
+  }
+
+  /**
+   * Sets xrp transaction.
+   *
+   * @param {rippleTypes.TransactionJSON} tx
+   */
+
+  setXRPTransaction(tx: rippleTypes.TransactionJSON): void {
+    this._xrpTransaction = tx;
```

Comment:
This should be in a constructor

Reply:
The constructor is based on `BaseTransaction`, only used to set the coinConfig